feature/google/src/test/java/BillingWrapperCommonTest.kt

```diff
@@ -55,98 +43,7 @@ import java.util.concurrent.CountDownLatch
-
-    private var mockPurchasesListener: BillingAbstract.PurchasesUpdatedListener = mockk()
-
-    private var capturedAcknowledgeResponseListener = slot<AcknowledgePurchaseResponseListener>()
-    private var capturedAcknowledgePurchaseParams = slot<AcknowledgePurchaseParams>()
-    private var capturedConsumeResponseListener = slot<ConsumeResponseListener>()
-    private var capturedConsumeParams = slot<ConsumeParams>()
-
-    private lateinit var wrapper: BillingWrapper
-
-    private lateinit var mockDetailsList: List<SkuDetails>
-
-    private var storeProducts: List<StoreProduct>? = null
-
-    private var skuDetailsResponseCalled = 0
-
-    private val billingClientOKResult = BillingClient.BillingResponseCode.OK.buildResult()
-    private val appUserId = "jerry"
-    private var mockActivity = mockk<Activity>()
-
-    @Before
-    fun setup() {
-        setupRunnables()
-
-        val listenerSlot = slot<PurchasesUpdatedListener>()
-        every {
-            mockClientFactory.buildClient(capture(listenerSlot))
-        } answers {
-            purchasesUpdatedListener = listenerSlot.captured
-            mockClient
-        }
-
-        val billingClientStateListenerSlot = slot<BillingClientStateListener>()
-        every {
-            mockClient.startConnection(capture(billingClientStateListenerSlot))
-        } answers {
-            billingClientStateListener = billingClientStateListenerSlot.captured
-        }
-
-        every {
-            mockClient.endConnection()
-        } just runs
-
-        val billingClientPurchaseHistoryListenerSlot = slot<PurchaseHistoryResponseListener>()
-        every {
-            mockClient.queryPurchaseHistoryAsync(
-                any(),
-                capture(billingClientPurchaseHistoryListenerSlot)
-            )
-        } answers {
-            billingClientPurchaseHistoryListener = billingClientPurchaseHistoryListenerSlot.captured
-        }
-
-        every {
-            mockClient.acknowledgePurchase(
-                capture(capturedAcknowledgePurchaseParams),
-                capture(capturedAcknowledgeResponseListener)
-            )
-        } just Runs
-
-        every {
-            mockClient.consumeAsync(capture(capturedConsumeParams), capture(capturedConsumeResponseListener))
-        } just Runs
-
-        every {
-            mockClient.isReady
-        } returns false andThen true
-
-        mockDetailsList = listOf(stubSkuDetails())
-
-        wrapper = BillingWrapper(mockClientFactory, handler, mockDeviceCache)
-        wrapper.purchasesUpdatedListener = mockPurchasesListener
-        onConnectedCalled = false
-        wrapper.stateListener = object : BillingAbstract.StateListener {
-            override fun onConnected() {
-                onConnectedCalled = true
-            }
-        }
-
-        every {
-            mockActivity.intent
-        } returns Intent()
-    }
+class BillingWrapperCommonTest : BillingWrapperTestBase() {
```

Comment:
This is the old, main, BillingWrapperTest, but with flavor-specific tests pulled out. It will be run for all flavors